lib/Dialect/RTL/RTLOps.cpp

```diff
@@ -459,6 +459,35 @@ static LogicalResult verifyInstanceOp(InstanceOp op) {
       return failure();
   }
 
+  // Check that result types are consistent with the referenced module.
+  auto expectedTypes = getModuleType(referencedModule).getResults();
+  auto numResults = op->getNumResults();
+
+  if (expectedTypes.size() != numResults) {
+    auto diag = op.emitOpError()
+                << "has a wrong number of results; expected "
+                << expectedTypes.size() << " but got " << numResults;
+    diag.attachNote(referencedModule->getLoc())
+        << "original module declared here";
+
+    return failure();
+  }
+
+  for (size_t i = 0; i != numResults; i++) {
```

Comment:
Super nit, prefer preincrement `++i` for consistency.

Reply:
addressed